src/components/settings/Calendars.js

```diff
@@ -203,14 +219,61 @@ export default class Calendars extends AddDeleteSetting {
     this.setState({ hexColor })
   }
 
+  handleFieInputChange = event => {
+    const { verifyNewCalendar } = this.props
+
+    console.log('selected files', event.target.files[0])
+    if (event.target.files[0]) {
+      let file = event.target.files[0]
+      let fileExt = file.name.split('.').pop()
+
+      if (fileExt.toLowerCase() !== 'ics') {
+        alert('Invalid file type')
+        return
+      }
+
+      let fileReader = new FileReader()
+      fileReader.onloadend = e => {
+        console.log('read file complete')
+
+        let calendar = {
+          uid: uuid(),
+          type: 'ics',
+          name: this.state.calendarName,
+          hexColor: this.state.hexColor,
+          mode: 'read-only',
+          data: {
+            src: fileReader.result,
+          },
+        }
+
+        verifyNewCalendar(calendar)
+      }
+
+      fileReader.readAsText(file)
+    }
+  }
+
+  handleAddCalendarClick = event => {
+    const { verifiedNewCalendarData, addItem } = this.props
+
+    if (verifiedNewCalendarData.status === 'ok') {
+      let calendar = Object.assign({}, verifiedNewCalendarData.calendar)
+      calendar.name = this.state.calendarName
+      calendar.hexColor = this.state.hexColor
+
+      addItem(calendar)
```

Comment:
`addItem` needs to handle `ics-raw` type

Reply:
I changed the importCalendarEvents function to handle with this.